### PR TITLE
8277853: With Touch enabled devices scrollbar disappears and the table is scrolled to the beginning

### PR DESCRIPTION
**Pull request by meghanEmbrace**
With a touch-enabled device, the scrollbar disappears a short while after it's used. During the layout, updateHbar() checks the hbar visibility and resets the clip, causing the user to be scrolled fully to the left when trying to access columns on the right. Using hbar.isVisible() is not feasible as there are times when the scrollbar is necessary but not visible (such as on touch-enabled devices where the scrollbar disappears when not in use, or when hidden by CSS). Hence, it is more reliable to use the variable that determines whether the hbar is necessary.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277853](https://bugs.openjdk.java.net/browse/JDK-8277853): With Touch enabled devices scrollbar disappears and the table is scrolled to the beginning

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Marius Hanl](https://openjdk.java.net/census#mhanl) (@Maran23 - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/688/head:pull/688` \
`$ git checkout pull/688`

Update a local copy of the PR: \
`$ git checkout pull/688` \
`$ git pull https://git.openjdk.java.net/jfx pull/688/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 688`

View PR using the GUI difftool: \
`$ git pr show -t 688`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/688.diff">https://git.openjdk.java.net/jfx/pull/688.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
This seems like the correct fix, but I'll ask @aghaisas to review. What testing have you done to ensure that there are no regressions or unintended side effects of this change? Can you provide an automated test?

**Comment by meghanEmbrace** [comment]
Added requested automated test.

**Review by kevinrushforth — COMMENTED** [review]
The test looks good with one comment about replacing the null test with an assert.

**Comment by meghanEmbrace** [comment]
Replaced the null test with an assert. Thanks.

**Review by kevinrushforth — APPROVED** [review]
The updated test looks good.

I verified that the new test fails without the fix and passes with the fix. This is a simple enough fix that it doesn't need a second reviewer. I can sponsor it once you integrate it.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewTest.java [hunk 1]
```diff
@@ -798,6 +798,37 @@ public void ensureRowRemainsSelectedWhenSelectingCellInSameRow() {
     /*********************************************************************
      * Tests for specific bugs                                           *
      ********************************************************************/
+    @Test public void test_jdk_8277853() {
```
Comment (Maran23) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewTest.java:801]: Not sure if there is some preference around here but I always like if the test method is not just named after the ticket but instead what it actually tests, e.g. something like `testInvisibleScrollbarDoesNotScrollTableToBeginning`. Optionally the ticket number can be referenced in the javadoc then. But that's just me.

Comment (kevinrushforth) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewTest.java:801]: I almost made the same comment as you did. In general we no longer use the pattern of putting the bug ID in the test name. In this case all of the surrounding tests do that, so I decided not to mention it, but I am perfectly happy for the test name to be changed to reflect its function (with the bug ID in a comment).

**Comment by meghanEmbrace** [comment]
Updated test name to be more descriptive.

**Review by kevinrushforth — APPROVED** [review]

**Comment by kevinrushforth** [comment]
This is now ready to be integrated.

**Review by Maran23 — APPROVED** [review]
Looks good to me too.
Tested with `-Dcom.sun.javafx.touch=true` and also without to make sure there is no regression.

**Pull request closed without merging**